providers/MongodbProvider.ts

```diff
@@ -1,44 +1,44 @@
-import { IocContract } from '@adonisjs/fold';
 import { ObjectId } from 'mongodb';
 
+import { ApplicationContract } from '@ioc:Adonis/Core/Application';
+
 import createMigration from '../src/Migration';
 import { Model, AutoIncrementModel } from '../src/Model/Model';
 import { Mongodb } from '../src/Mongodb';
 
 export default class MongodbProvider {
-  private $container: IocContract;
-
-  public constructor(container: IocContract) {
-    this.$container = container;
-  }
+  public static needsApplication = true;
+  public constructor(protected app: ApplicationContract) {}
 
   public register(): void {
-    this.$container.singleton('Mongodb/Database', () => {
-      const config = this.$container
+    this.app.container.singleton('Mongodb/Database', () => {
+      const config = this.app.container
         .use('Adonis/Core/Config')
         .get('mongodb', {});
-      const Logger = this.$container.use('Adonis/Core/Logger');
+      const Logger = this.app.container.use('Adonis/Core/Logger');
       return new Mongodb(config, Logger);
```

Comment:
here you can use `this.app.config` and `this.app.logger`

Reply:
`this.app.logger` is a LoggerContract, type that is not compatible with `@poppinss/fancy-logs` Logger.
Moreover, `@poppinss/fancy-logs`'s Logger is not the same as `@poppinss/cliui`'s one.